### ISSUE TITLE
TLS 1.3 - newbie question

### ISSUE THREAD

**Issue by MSP-Greg** [jruby/jruby-openssl#184]
Some gems are implementing compatibility with TLS 1.3.  One (parsed) example of code I've seen is below, removing TLS 1.0 from allowable protocols:

```java
engine = sslCtx.createSSLEngine();
protocols = new String[] { "TLSv1.1", "TLSv1.2" };
engine.setEnabledProtocols(protocols);
```

Is TLS 1.3 available on java , and if so, how can one test for it in code similar to the above?  Or, can one just add it to protocols without generating an error?

Sorry for the dumb question, not sure where to ask...

**Comment by ahorek** [comment]
_Is TLS 1.3 available on java?_
java 11+ https://bugs.openjdk.java.net/browse/JDK-8202625

I think that just adding a new protocol won't be enough, there're also new ciphers...

**Comment by MSP-Greg** [comment]
Thanks for the response.  I'm sorry I wasn't clearer.

What I'm wondering about is how to query what protocols are available when using JRuby.

For instance, Puma's 'MRI' CI is (deliberately) testing using OpenSSL 1.0.1, 1.0.2, and 1.1.1.  Some don't have TLSv1.3, and some may be built without SSLv3 (and SSLv2).

Puma rejects SSLv3 connections, regardless of the OS's capabilities.  If we're testing on an OS without SSLv3, we'd prefer to show an 'SSLv3 rejection' test as a skip, as we can't generate an SSLv3 connection for the client.  If we can create a client with an SSLv3 connection, the test should pass.  We will have some TLSv1.3 specific tests also.

**Comment by kares** [comment]
TLS 1.3 is available since 0.12.0 (on Java >= 8)
